# TASK 1: Exploratory Data Analysis (EDA)

## Objective
The objective of this task is to explore the dataset, understand its structure, identify patterns, detect anomalies, validate assumptions, and discover potential data quality issues before performing advanced analysis.

Dataset Used: Amazon Product Reviews Dataset

## TASK_1.1 - Ask Meaningful Questions.


In [1]:
print("Question 1: What is the distribution of review scores?")
print("Question 2: Are positive reviews more common than negative reviews?")
print("Question 3: Are there any missing values in the dataset?")
print("Question 4: What is the average review rating?")
print("Question 5: Are there duplicate reviews present?")

Question 1: What is the distribution of review scores?
Question 2: Are positive reviews more common than negative reviews?
Question 3: Are there any missing values in the dataset?
Question 4: What is the average review rating?
Question 5: Are there duplicate reviews present?


## Task_1.2 - Explore Dataset Structure, Variables and Data Types.

In [2]:
import pandas as pd

# Load Dataset
df = pd.read_csv("Reviews.csv")

print("Dataset Shape:", df.shape)

print("\nColumn Names:")
print(df.columns)

print("\nData Types:")
print(df.dtypes)

df.head()


Dataset Shape: (568454, 10)

Column Names:
Index(['Id', 'ProductId', 'UserId', 'ProfileName', 'HelpfulnessNumerator',
       'HelpfulnessDenominator', 'Score', 'Time', 'Summary', 'Text'],
      dtype='str')

Data Types:
Id                        int64
ProductId                   str
UserId                      str
ProfileName                 str
HelpfulnessNumerator      int64
HelpfulnessDenominator    int64
Score                     int64
Time                      int64
Summary                     str
Text                        str
dtype: object


,Id,ProductId,UserId,ProfileName,HelpfulnessNumerator,HelpfulnessDenominator,Score,Time,Summary,Text
0,1,B001E4KFG0,A3SGXH7AUHU8GW,delmartian,1,1,5,1303862400,Good Quality Dog Food,I have bought several of the Vitality canned d...
1,2,B00813GRG4,A1D87F6ZCVE5NK,dll pa,0,0,1,1346976000,Not as Advertised,Product arrived labeled as Jumbo Salted Peanut...
2,3,B000LQOCH0,ABXLMWJIXXAIN,"Natalia Corres ""Natalia Corres""",1,1,4,1219017600,"""Delight"" says it all",This is a confection that has been around a fe...
3,4,B000UA0QIQ,A395BORC6FGVXV,Karl,3,3,2,1307923200,Cough Medicine,If you are looking for the secret ingredient i...
4,5,B006K2ZZ7K,A1UQRSCLF8GW1T,"Michael D. Bigham ""M. Wassir""",0,0,5,1350777600,Great taffy,Great taffy at a great price. There was a wid...


## Task_1.3 - Identify Trends, Patterns and Anomalies.

In [3]:
print("Review Score Distribution:\n")
print(df["Score"].value_counts().sort_index())

print("\nAverage Rating:")
print(round(df["Score"].mean(), 2))

Review Score Distribution:

Score
1     52268
2     29769
3     42640
4     80655
5    363122
Name: count, dtype: int64

Average Rating:
4.18


## Task_1.4 - Test Hypothesis and Validate Assumptions.

In [4]:
positive_reviews = len(df[df["Score"] >= 4])
negative_reviews = len(df[df["Score"] <= 2])

print("Positive Reviews:", positive_reviews)
print("Negative Reviews:", negative_reviews)

if positive_reviews > negative_reviews:
    print("\nHypothesis Validated: Positive reviews are more common.")
else:
    print("\nHypothesis Rejected.")

Positive Reviews: 443777
Negative Reviews: 82037

Hypothesis Validated: Positive reviews are more common.


## Task_1.5 - Detect Data Issues and Problems.

In [5]:
print("Missing Values:\n")
print(df.isnull().sum())

print("\nDuplicate Records:")
print(df.duplicated().sum())

Missing Values:

Id                         0
ProductId                  0
UserId                     0
ProfileName               26
HelpfulnessNumerator       0
HelpfulnessDenominator     0
Score                      0
Time                       0
Summary                   27
Text                       0
dtype: int64

Duplicate Records:
0


# TASK_1 - SUMMARY

### Key Findings

- Successfully loaded and explored the Amazon Product Reviews dataset.
- Dataset contains 568,454 records and 10 features.
- Data types and column structure were examined to understand the dataset.
- Review score distribution analysis showed that ratings of 4 and 5 are the most common.
- The average product rating was found to be 4.18, indicating generally positive customer feedback.
- Positive reviews significantly outnumber negative reviews, validating the hypothesis that customers are mostly satisfied with the products.
- Missing values were detected in the ProfileName and Summary columns.
- No duplicate records were found in the dataset.
- Overall, the dataset is clean and suitable for further visualization and sentiment analysis.

### Conclusion

The Exploratory Data Analysis (EDA) helped in understanding the dataset structure, identifying patterns in customer ratings, validating assumptions, and detecting potential data quality issues. The dataset is ready for the next stage of analysis and visualization.